### DISCUSSION TITLE
Regarding /spawnboss command

### DISCUSSION THREAD

**Discussion by moisterrific** [Pryaxis/TShock#1942] [Ideas]
Should all “boss” type NPCs be added to /spawnboss? 

As in all the ones that display a health bar at the bottom of the screen? Or exclude mini-ish bosses like The Flying Dutchman and Martian Mothership?

**Comment by ronelm2000** [comment]
They ... wouldn't drop anything anyway as long as it's not the appropriate invasion.

**Reply by moisterrific** [reply]
Never knew that, I thought the drops aren’t affected by event. Cuz right now I’m thinking of adding all NPCs with “boss” health bars and purple “…had been defeated!” messages into the command.

**Reply by Kojirremer** [reply]
AFAIK, the ones without drops unless under their own event are Pumpkin and Frost Moon bosses, and maybe DD2's. Minibosses from normal invasions drop their junk anytime, but will now actively try to despawn (they didn't prior to 1.3).

**Comment by hakusaro** [comment]
I don't really see a reason why not

**Comment by bartico6** [comment]
I think adding any "gameplay significant" entity to this command makes sense even if its not an actual boss. Spawning Dutchmen, Ice Queens, Pumpkings etc makes total sense to me for this command. Anything miniboss+ is absolutely sensible to be in /spawnboss in my opinion

**Comment by hakusaro** [comment]
I tend to agree.

On May 16, 2021, GitHub ***@***.***> wrote:
> I think adding any "gameplay significant" entity to this command makes
> sense even if its not an actual boss. Spawning Dutchmen, Ice Queens,
> Pumpkings etc makes total sense to me for this command. Anything
> miniboss+ is absolutely sensible to be in /spawnboss in my opinion
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Pryaxis/TShock/discussions/1942#discussioncomment-
745553>, or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAECBJ4CMTMG4XOD4MNLQO3TOAQWJANCNFSM4NNTDXTA>.

**Reply by Kojirremer** [reply]
But would they also call their event, like nocturnal bosses do? Because as stated above, pumpkin/frost bosses have no loot unless on their event, and things like Mothron and Dreadnaughtilus despawn immediately if not spawned during an eclipse/bloodmoon.

**Reply by bartico6** [reply]
This is an interesting question - tShock has historically [made sure that boss spawning won't fail due to spawn requirements](https://github.com/Pryaxis/TShock/blob/general-devel/TShockAPI/Commands.cs#L2438) but summoning an entire event/invasion feels somewhat excessive.

**Comment by hakusaro** [comment]
We tend to believe that a “command” to do something shouldn’t be
overruled by game logic. The goal is to force certain things that would
otherwise be impossible to be possible. Hence why we had teleportation
using an elaborate spawn point shuffle routine for a while.

On May 16, 2021, GitHub ***@***.***> wrote:
> This is an interesting question - tShock has historically made sure
> that boss spawning won't fail due to spawn requirements
> <https://github.com/Pryaxis/TShock/blob/general-
> devel/TShockAPI/Commands.cs#L2438> but summoning an entire
> event/invasion feels somewhat excessive.
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Pryaxis/TShock/discussions/1942#discussioncomment-
745698>, or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAECBJ26FZK3R7DXQEXPZWLTOA26NANCNFSM4NNTDXTA>.
